### DISCUSSION TITLE
Welcome to pi4j-v2 Discussions!

### DISCUSSION THREAD

**Discussion by FDelporte** [Pi4J/pi4j#58] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by hackerjimbo** [comment]
Hi Frank,

Good idea to start the discussions going.

I'm Jim and I've been working on getting device drivers written for many of the Piromoni boards (as well as some others). Right now I'm working on integrating their Touch PHAT with it's capacitive buttons. As a result of this I'm having to create a new DigitalInputProvider which is proving a little… interesting. I'm thinking that as soon as an instance of the device itself is created (for which you need an I2C bus/address pair it should create an register a new Provider that in turn can allow a DigitalInput to be created. Then I need to figure out how to add listeners to it.

There are a lot of base classes that make it simpler the add the default methods but I'm still working my way though it. For example, adding a listener seems to be mostly handled by some superclasses. I just need to let them know of a state change and they do all the good work.

I'm also looking forward to Frank's talk at the upcoming FOSDEM.

**Reply by FDelporte** [reply]
Hey, just setting up the discussion part as a test to see if this could be good replacement for the forums. And you are the first contributor! ;-)
Thanks for your support and work! Work is ongoing to get the V1.4 released, so we can than further focus on the V2 and how we want to organize and implement device support.
